protocol/x/subaccounts/keeper/subaccount_test.go

```diff
@@ -2260,6 +2261,430 @@ func TestUpdateSubaccounts(t *testing.T) {
 	}
 }
 
+func TestUpdateSubaccounts_WithdrawalsBlocked(t *testing.T) {
```

Comment:
A few notes here:
- I'm planning on only adding tests for `UpdateSubaccounts` here, since the only logic is modified in `internalCanUpdateSubaccounts` (which is shared between `UpdateSubaccounts` and `CanUpdateSubaccounts`).
    - Note: this means that technically this logic isn't tested for things like add-to-orderbook collateralization checks, which _should_ be fine, but LMK if anyone disagrees.
- I still need to add tests for transfers and matches (they'll look very similar to the current tests for withdrawals and deposits respectively).
- I am not adding E2E tests in this PR for withdrawal gating, and will do that when the feature is fully implemented (requires `PrepareCheckState` changes for adding the zero-fill deleveraging matches and `DeliverTx` changes for writing the block number to state when negative TNC subaccount seen).

Reply:
Added tests for transfers and matches!